adapters/rest-vertx/src/main/java/org/eclipse/hono/adapter/rest/credentials/VertxRestUser.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.adapter.rest.credentials;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AbstractUser;
+import io.vertx.ext.auth.AuthProvider;
+import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.CredentialsConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper class to handle http basic authentication.
+ * <p>
+ * The tenant is transferred as part of the username in the basic authentication header similar to the MQTT
+ * username/password implementation.
```

Comment:
General remark:
in mqtt we do not have custom headers, so we needed a solution to carry the tenantId and chose the username to contain the tenantId AND the username.
In http, we have any number of custom headers - what about using a header for the tenantId in this protocol?
Usually a client belongs to only one tenant and only the username varies IMHO, so there would not be the need to construct an artificial username on clientside that is parsed again on the server side.
Might be not so straightforward to access the header in the code, but I think it should be possible.
WDYT?